### PR TITLE
Reimplement search route

### PR DESCRIPTION
**Pull request by manuelroth**
- This PR reimplements the search route but keeps the old route for now
- The new route takes requests in the following form:
```http
GET /search?
tool=imageslider&
createdBy=manuel.roth@nzz.ch&
department=Storytelling%20/%20Infografik&
publication=nzz&
limit=18
```
- The results are sorted by field `updatedDate`

### CODE REVIEW

File: plugins/core/db/index.js [hunk 1]
```diff
@@ -1,11 +1,63 @@
-const nano = require('nano');
-const Boom = require('boom');
+const nano = require("nano");
+const Boom = require("boom");
+
+function getFilters(queryParameters) {
```
Comment (benib) [inline plugins/core/db/index.js]: please name this function `getSearchFilters` as this is only covering the fixed set of search query parameters we support and is not a generic function to generate query parameters to filters for mango.

File: plugins/core/db/index.js [hunk 2]
```diff
@@ -1,11 +1,63 @@
...
+  let filters = [];
+  for (let parameter in queryParameters) {
+    if (
+      queryParameters[parameter] &&
+      !Array.isArray(queryParameters[parameter]) &&
+      parameter !== "searchString" &&
+      parameter !== "bookmark" &&
+      parameter !== "limit"
+    ) {
+      let fieldFilter = {};
```
Comment (benib) [inline plugins/core/db/index.js]: use `const` here as this is never being reassigned.

File: plugins/core/db/index.js [hunk 3]
```diff
@@ -1,11 +1,63 @@
-const nano = require('nano');
-const Boom = require('boom');
+const nano = require("nano");
+const Boom = require("boom");
+
+function getFilters(queryParameters) {
+  let filters = [];
+  for (let parameter in queryParameters) {
+    if (
```
Comment (benib) [inline plugins/core/db/index.js]: I'd prefer an implementation like this:

```
filters = Object.keys(queryParameters)
  .map(parameterName => {
    if (parameterName === "searchString") {
       const filter = ....
       return filter;
    }
    if (other special case) {
      const filter = ...
      return filter;
    }
    // all the other special cases
    // the default case
  })
```

this results in smaller if blocks. returning early makes it much easier to read imo as there is no need for `else`.
the early returns also help to make it very clear that there is no other path the code could take later on in the function.

**Comment by benib** [comment]
@manuelroth can you update the description of the PR to reflect the recent changes regarding the result ordering?

also: imho the old POST /search route can be dropped and we do a major release with this one. no need to keep that.

File: plugins/core/base/routes/search.js [hunk 4]
```diff
@@ -1,16 +1,33 @@
...
-    tags: ['api', 'editor']
+    tags: ["api", "editor"]
   },
   handler: async (request, h) => {
-    return request.server.methods.db.item.search(request.payload);
+    // Creates new object filterProperties which contains all properties but bookmark and limit
+    const { bookmark, limit, ...filterProperties } = request.query;
+    return request.server.methods.db.item.search(
+      filterProperties,
+      request.query.limit,
```
Comment (benib) [inline plugins/core/base/routes/search.js]: can you use `limit` and `bookmark` here as it's already extracted from `request.query`. Makes it more clear imo.

Comment (manuelroth) [reply plugins/core/base/routes/search.js]: Sure, fixed in https://github.com/nzzdev/Q-server/pull/111/commits/cdcbe7073ae3938d278d9779acca17b0121c2695

**Comment by benib** [comment]
this will be in the next release.

**Pull request merged**